### PR TITLE
Use a PDB if the Enabled in the values.yaml

### PR DESCRIPTION
**Pull request by justdan96**
This is a pretty simple PR to the Helm chart, just adding a PDB if there is more than one replica specified in the values.yaml.

### CODE REVIEW

**Comment by justdan96** [comment]
Linting output:
```
% ct lint --validate-maintainers=false --all
Linting charts...
Version increment checking disabled.

------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 nfs-subdir-external-provisioner => (version: "4.0.18", path: "charts/nfs-subdir-external-provisioner")
------------------------------------------------------------------------------------------------------------------------

Linting chart "nfs-subdir-external-provisioner => (version: \"4.0.18\", path: \"charts/nfs-subdir-external-provisioner\")"
Validating /Users/daniel.bryant/Code/nfs-subdir-external-provisioner/charts/nfs-subdir-external-provisioner/Chart.yaml...
Validation success! 👍

Linting chart with values file "charts/nfs-subdir-external-provisioner/ci/test-values.yaml"...

==> Linting charts/nfs-subdir-external-provisioner
[INFO] Chart.yaml: icon is recommended
[WARNING] templates/podsecuritypolicy.yaml: policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+

1 chart(s) linted, 0 chart(s) failed
------------------------------------------------------------------------------------------------------------------------
 ✔︎ nfs-subdir-external-provisioner => (version: "4.0.18", path: "charts/nfs-subdir-external-provisioner")
------------------------------------------------------------------------------------------------------------------------
All charts linted successfully
```

File: charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{{ if gt .Values.replicaCount 1.0}}
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml:1]: 1. there is no reason to use float 1.0, 1 is enough and add space between 1 and `}}`
2. there need be a value to enable/disable PDB
3. nit: it is recommended to use `{{-` in such `if`s and `end`s

File: charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+apiVersion: policy/v1
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml:2]: nfs-subdir-external-provisioner currently supports Kubernetes >=1.9. PodDisruptionBudget was beta from 1.5 to 1.20 and GA only in v1.21. so up to 1.20 the `apiVersion` is `policy/v1beta1`.

- v1.5 - v1.20: policy/v1beta1
- v1.21 : policy/v1

File: charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+kind: PodDisruptionBudget
+metadata:
+  labels:
+    {{- include "nfs-subdir-external-provisioner.labels" . | nindent 4 }}
+  name: {{ template "nfs-subdir-external-provisioner.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  maxUnavailable: 1
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml:10]: maxUnavailable should be configurable from values

File: charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{{ if gt .Values.replicaCount 1.0}}
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  labels:
+    {{- include "nfs-subdir-external-provisioner.labels" . | nindent 4 }}
+  name: {{ template "nfs-subdir-external-provisioner.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml:8]: explicitly set the namespace is optional and can be omitted

**Review by yonatankahana — CHANGES_REQUESTED** [review]

**Comment by justdan96** [comment]
I'll look at fixing those issues you mentioned on Monday.

**Comment by justdan96** [comment]
@yonatankahana let me know if those amendments look okay to you

**Comment by justdan96** [comment]
@yonatankahana / @jackielii can you please let me know if all is good with the PR now and it can be merged?

File: charts/nfs-subdir-external-provisioner/values.yaml [hunk 5]
```diff
@@ -108,3 +108,7 @@ affinity: {}
 
 # Additional labels for any resource created
 labels: {}
+
+podDisruptionBudget:
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/values.yaml:112]: values should be documented in the [chart readme](https://github.com/kubernetes-sigs/nfs-subdir-external-provisioner/blob/master/charts/nfs-subdir-external-provisioner/README.md#configuration)

**Review by yonatankahana — CHANGES_REQUESTED** [review]

File: charts/nfs-subdir-external-provisioner/values.yaml (hunk 5 above)
Comment (justdan96) [reply charts/nfs-subdir-external-provisioner/values.yaml:112]: these have now been added

**Comment by justdan96** [comment]
@yonatankahana I've now updated the chart readme, please let me know if anything else is required

File: charts/nfs-subdir-external-provisioner/README.md [hunk 6]
```diff
@@ -85,6 +85,8 @@ The following tables lists the configurable parameters of this chart and their d
 | `affinity`                          | Affinity settings                                                                                     | `{}`                                                     |
 | `tolerations`                       | List of node taints to tolerate                                                                       | `[]`                                                     |
 | `labels`                            | Additional labels for any resource created                                                            | `{}`                                                     |
+| `podDisruptionBudget.enabled`       | Create & use Pod Disruption Budget                                                                    | `false`                                               |
+| `podDisruptionBudget.maxUnavailable`| Set maximum unavailable pods in the Pod Disruption Budget                                             | `1`                                                  |
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/README.md:89]: there's need to be a space after the backquote. also make sure you keep the table aligned

File: charts/nfs-subdir-external-provisioner/README.md [hunk 7]
```diff
@@ -85,6 +85,8 @@ The following tables lists the configurable parameters of this chart and their d
 | `affinity`                          | Affinity settings                                                                                     | `{}`                                                     |
 | `tolerations`                       | List of node taints to tolerate                                                                       | `[]`                                                     |
 | `labels`                            | Additional labels for any resource created                                                            | `{}`                                                     |
+| `podDisruptionBudget.enabled`       | Create & use Pod Disruption Budget                                                                    | `false`                                               |
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/README.md:88]: please use the word 'and' instead of ampersand symbol

**Review by yonatankahana — CHANGES_REQUESTED** [review]

**Comment by justdan96** [comment]
@yonatankahana can this PR be merged now?

File: charts/nfs-subdir-external-provisioner/README.md [hunk 8]
```diff
@@ -97,3 +99,9 @@ helm install second-nfs-subdir-external-provisioner nfs-subdir-external-provisio
     --set storageClass.name=second-nfs-client \
     --set storageClass.provisionerName=k8s-sigs.io/second-nfs-subdir-external-provisioner
 ```
+
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/README.md:102]: remove the extra lines please

File: charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{{- if .Values.podDisruptionBudget.enabled }}
+apiVersion: {{ template "podDisruptionBudget.apiVersion" . }}
+kind: PodDisruptionBudget
+metadata:
+  labels:
+    {{- include "nfs-subdir-external-provisioner.labels" . | nindent 4 }}
+  name: {{ template "nfs-subdir-external-provisioner.fullname" . }}
+spec:
+    maxUnavailable: {{ .Values.podDisruptionBudget.maxUnavailable | default 1 }}
```
Comment (yonatankahana) [inline charts/nfs-subdir-external-provisioner/templates/poddisruptionbudget.yaml:9]: wrong indentation

**Review by yonatankahana — CHANGES_REQUESTED** [review]

**Comment by yonatankahana** [comment]
/approve
/lgtm

Thank you for your contribution @justdan96 :+1:

**Pull request merged**